src/object/player.cpp

```diff
@@ -360,6 +367,58 @@ Player::update(float dt_sec)
 
   // extend/shrink tux collision rectangle so that we fall through/walk over 1
   // tile holes
+  
+  //wallclinging and walljumping
+  
+  Rectf wallclingleft = get_bbox();
+  wallclingleft.set_left(wallclingleft.get_left() - 8);
+  bool trygrabwallleft = !Sector::get().is_free_of_statics(wallclingleft);
+  m_on_left_wall = ((trygrabwallleft) ? true : false);
+
```

Comment:
The code is literally perfect in my eyes except for this line.

This line.

_screeches loudly_

--------------------------

This line (and another one 5-ish lines below) could be simplified to `m_on_left_wall = trygrabwallleft;`
Considering this would make both variables identical, do you need `trygrabwallleft`? Can you replace it by `m_on_left_wall`?